test/test.js

```diff
@@ -909,6 +930,271 @@ test('Should throw if both path and url are missing', (t) => {
   }
 })
 
+test('chainable api: backwards compatibility for promise (then)', (t) => {
+  t.plan(1)
+
+  const dispatch = function (req, res) {
+    res.writeHead(200, { 'Content-Type': 'text/plain' })
+    res.end('hello')
+  }
+
+  inject(dispatch)
+    .get('/')
+    .then(res => t.equal(res.payload, 'hello'))
+    .catch(err => t.fail(err))
+})
+
+test('chainable api: backwards compatibility for promise (catch)', (t) => {
+  t.plan(1)
+
+  function dispatch (req, res) {
+    throw Error
+  }
+
+  inject(dispatch)
+    .get('/')
+    .catch(err => t.ok(err))
+})
+
+test('chainable api: multiple call of then should returns the same promise', (t) => {
+  t.plan(1)
+
+  function dispatch (req, res) {
+    res.writeHead(200, { 'Content-Type': 'text/plain' })
```

Comment:
Can you also add a check to be 100% sure that dispatch has been called only once?